compiler/expressions.go

```diff
@@ -592,7 +592,7 @@ func (c *funcContext) translateExpr(expr ast.Expr) *expression {
 				declaredFuncRecv := sel.Obj().(*types.Func).Type().(*types.Signature).Recv().Type()
 				if typesutil.IsJsObject(declaredFuncRecv) {
 					globalRef := func(id string) string {
-						if recv.String() == "$global" && id[0] == '$' {
+						if recv.String() == "$global" && id[0] == '$' && len(id) > 1 {
```

Comment:
I see, that makes sense.

Still, if it [were me](https://github.com/gopherjs/gopherjs/issues/374#issuecomment-170292266), I would swap the order of `len(id) > 1` and `id[0] == '$'` checks. The reason to do so is, code is the same, but it becomes easier for myself and others to read and be able to tell at a glance it won't panic.

The current code requires either being very familiar with the context already, or spending more time reading the surrounding code and trying to come to the conclusion that `id` will never be passed to `globalRef` as an empty string. I don't see any advantages to checking `id[0] == '$'` before `len(id) > 1`, are there any?

This is a very minor code review comment.